### DISCUSSION TITLE
Why does this library outputs to ONNX and tensorflow if Pytorch is still needed?

### DISCUSSION THREAD

**Discussion by DavidGOrtega** [snakers4/silero-models#209] [Q&A]
👋 Having looked into the examples I have seen that despite that the model is converted into ONNX or TF you still need the Decoder in Pytorch. What is the benefit of the conversion then?

**Comment by snakers4** [comment]
If the question is about the stt model, the decoder does not rely on PyTorch.

**Reply by DavidGOrtega** [reply]
Thanks for the reply @snakers4 🙏 
What I meant is that despite the effort of transforming the model into ONNX or TF some parts like the Decoder still relies in Pytorch. Or at least that what I can infer.

Within the current onnx example:

```python
_, decoder, utils = torch.hub.load(repo_or_dir='snakers4/silero-models', model='silero_stt', language='en')
```

Then de Decoder class has Torch dependency like i.e.:

```python
argm = torch.argmax(probs, axis=1)
```

Ideally the model should work completely with ONNX alone? If not why should I run the stt model in ONNX and not in Pytorch?

**Reply by snakers4** [reply]
Well, I guess it could depend on numpy's argmax. We just did not really invest into this, because the users most likely will rewrite the decoder anyaway.

**Reply by DavidGOrtega** [reply]
> because the users most likely will rewrite the decoder anyaway.

Why? 🤔
